pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -1618,6 +1720,84 @@ func doAppNetworkModifyUnderlayNetwork(
 	publishNetworkInstanceStatus(ctx, netstatus)
 
 	maybeRemoveStaleIpsets(staleIpsets)
+	return nil
+}
+
+// returns a interfaces list, with network change
+func doAppNetworkModifyUNetAppNums(
+	ctx *zedrouterContext,
+	config types.AppNetworkConfig,
+	oldConfig types.AppNetworkConfig,
+	status *types.AppNetworkStatus,
+	ipsets []string) ([]bool, error) {
+	var intfList []bool
+	for i := range config.UnderlayNetworkList {
+		log.Tracef("handleModify ulNum %d\n", i)
+		ulConfig := &config.UnderlayNetworkList[i]
+		oldulConfig := &oldConfig.UnderlayNetworkList[i]
+		ulStatus := &status.UnderlayNetworkList[i]
+		change, err := doAppNetworkModifyUNetAppNum(ctx,
+			status, ulConfig, oldulConfig, ulStatus, ipsets)
+		if err != nil {
+			return intfList, err
+		}
+		intfList = append(intfList, change)
+	}
+	publishAppNetworkStatus(ctx, status)
+	return intfList, nil
+}
+
+// on network transition,
+// if activated, deactivate the underlay interface
```

Comment:
Clearly this comment is incorrect since it only does the if it was a change, but see comment below.